setup.py

```diff
@@ -60,16 +61,28 @@
     compiler_directives['linetrace'] = True
     define_macros.append(("CYTHON_TRACE_NOGIL", "1"))
 
+
+if sys.platform == "darwin":
+    os.environ["CC"] = "/usr/local/opt/llvm/bin/clang -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib"
+    os.environ["CXX"] = "/usr/local/opt/llvm/bin/clang++ -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib"
```

Comment:
What is happening if we run setup on a Mac without llvm installed at this exact location?
This path works for Travis CI on Mac, but we will need to make it compatible with our Jenkins build also, and potentially with other machines where the users want to use the library from sources.
Could you think of some ENV vars to use? Or a fallback or not building with OMP in case these are missing ?

Reply:
On our mac build machine, for example, we have installed `/opt/clang370/OpenMP/lib`

Reply:
Looks like homebrew always keeps a symlink at `/usr/local/opt/` ([ref](https://osxdaily.com/2018/07/05/where-homebrew-packages-installed-location-mac/)). So, if `brew install` is successful, we will always have clang in that particular location.

> 
> 
> On our mac build machine, for example, we have installed `/opt/clang370/OpenMP/lib`

This is probably because in the Jenkins machine the version of the clang compiler is updated (clang 3.7) where OpenMP is supported.

Reply:
Indeed. Could we then try to make this script more generic?
For example   define `os.environ["CC"]` only in case it is not already defined on that Mac, or it does not contain already `clang` in it ?

Reply:
The "CC" environmental variable stores the location of the C++ compiler. Even though if it contains `clang`, we cannot be sure if there is OpenMP installed.

Reply:
Then let me phrase it differently:
- the current hard-coded paths in the `setup.py` are not ok
- I do not have a final solution how to fix that part. Maybe move the env vars set for CC and CXX in `travis.yml` and check (plus document) that the users of the sources will need to do the same ?

Reply:
Done.